app/experimenter/nimbus-ui/src/components/PageResults/index.tsx

```diff
@@ -108,46 +117,159 @@ const PageResults: React.FunctionComponent<RouteComponentProps> = () => {
+                            key={metric.slug}
+                            outcomeSlug={metric.slug!}
+                            outcomeDefaultName={metric?.friendlyName!}
+                            metricType={METRIC_TYPE.PRIMARY}
+                          />
+                          <AnalysisErrorAlert
+                            errors={analysis.errors[metric.slug]}
+                          />
+                        </>
+                      );
+                    }
+                  }
+                  return (
+                    <TableMetricCount
+                      key={metric?.slug}
+                      outcomeSlug={metric?.slug!}
+                      outcomeDefaultName={metric?.friendlyName!}
+                      group={GROUP.OTHER}
+                      metricType={METRIC_TYPE.PRIMARY}
+                    />
+                  );
+                });
+              })
+            : // no Overall results, check for errors in primary outcome metrics
+              experiment.primaryOutcomes?.map((slug) => {
+                const outcome = configOutcomes!.find((set) => {
+                  return set?.slug === slug;
+                });
+                return outcome?.metrics?.map((metric) => {
+                  if (
+                    metric?.slug &&
+                    analysis?.errors &&
+                    Object.keys(analysis.errors).length > 1
+                  ) {
+                    // if we have more than just the 'experiment' errors key,
+                    // that means at least one metric had errors, so we should
+                    // show all the metric headers to avoid potential confusion
+                    const metricHeader = (
+                      <MetricHeader
+                        key={metric.slug}
+                        outcomeSlug={metric.slug!}
+                        outcomeDefaultName={metric?.friendlyName!}
+                        metricType={METRIC_TYPE.PRIMARY}
+                      />
+                    );
+                    if (
+                      metric.slug in analysis.errors &&
+                      analysis.errors[metric.slug].length > 0
+                    ) {
+                      primaryOutcomeMetricsWithErrors.push(metric.slug);
+                      return (
+                        <>
+                          {metricHeader}
+                          <AnalysisErrorAlert
+                            errors={analysis.errors[metric.slug]}
+                          />
+                        </>
+                      );
+                    } else {
+                      return (
+                        <>
+                          {metricHeader}
+                          <p>
+                            <i>No results available for metric.</i>
+                          </p>
+                        </>
+                      );
+                    }
+                  }
+                });
+              })}
```

Comment:
If you have to do this whole same block for primary and secondary outcomes, couldn't you just pull it out into a function?